callouts/python/extproc/service/callout_server.py

```diff
@@ -61,23 +63,104 @@ def add_header_mutation(
   return header_mutation
 
 
+def update_header_mutation(
+    headers: service_pb2.HttpHeaders,
+    update: list[tuple[str, str]] | None = None,
+    clear_route_cache: bool = False,
+) -> service_pb2.HeadersResponse:
+  """Generate a header response for incoming requests.
+
+  Args:
+    headers: Current headers presented in the request
+    update: A list of tuples representing headers to update.
+    clear_route_cache: If true, will enable clear_route_cache on the response.
+
+  Returns:
+    The constructed header response object.
+  """
+
+  header_mutation = service_pb2.HeadersResponse()
+  add_dict = {k: bytes(v, 'utf-8') for k, v in update} if update else {}
+
+  for header in headers.headers.headers:
+    # Convert existing header value to bytes if it's not already
+    existing_value = header.raw_value if isinstance(header.raw_value, bytes) else bytes(header.raw_value, 'utf-8')
+
+    # Use the new value from add_dict if it exists, otherwise use the existing header value
+    new_value = add_dict.get(header.key, existing_value)
+    header_mutation.response.header_mutation.set_headers.append(
+      service_pb2.HeaderValueOption(
+        header=service_pb2.HeaderValue(key=header.key, raw_value=new_value),
+        append_action=3  # OVERWRITE_IF_EXISTS
+      )
+    )
+
+  if clear_route_cache:
+    header_mutation.response.clear_route_cache = True
+  return header_mutation
+
+
+def normalize_header_mutation(
```

Comment:
There is no real conception of updating the headers from the callout in terms of what is represented in the proto.

would it be better to consolidate this update method into the add method, adding the ability to set the append action?

Something like this:

```
def add_header_mutation(
    add: list[tuple[str, str]] | None = None,
    remove: list[str] | None = None,
    clear_route_cache: bool = False,
    append_action: service_pb2.HeaderValueOption.HeaderAppendAction = None,
) -> service_pb2.HeadersResponse:
  """Generate a header response for incoming requests.

  Args:
    add: A list of tuples representing headers to add.
    remove: List of header strings to remove from the request.
    clear_route_cache: If true, will enable clear_route_cache on the response.

  Returns:
    The constructed header response object.
  """
  header_mutation = service_pb2.HeadersResponse()
  if add:
    for k, v in add:
      header_value_option = service_pb2.HeaderValueOption(
          header=service_pb2.HeaderValue(key=k, raw_value=bytes(v, 'utf-8'))
        )
      if append_action:
        header_value_option.append_action = append_action
      header_mutation.response.header_mutation.set_headers.append(header_value_option)
  if remove is not None:
    header_mutation.response.header_mutation.remove_headers.extend(remove)
  if clear_route_cache:
    header_mutation.response.clear_route_cache = True
  return header_mutation
```